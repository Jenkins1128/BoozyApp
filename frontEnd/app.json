{
	"expo": {
		"name": "Boozy",
		"slug": "Boozy",
		"version": "1.0.0",
		"orientation": "portrait",
		"icon": "./assets/icon.png",
		"splash": {
			"image": "./assets/splash.png",
			"resizeMode": "contain",
			"backgroundColor": "#ffffff"
		},
		"updates": {
			"fallbackToCacheTimeout": 0
		},
		"assetBundlePatterns": ["**/*"],
		"ios": {
			"supportsTablet": true,
			"bundleIdentifier": "com.Isaiah-Jenkins.Boozy"
		},
		"android": {
			"package": "com.IsaiahJenkins.Boozy",
			"adaptiveIcon": {
				"foregroundImage": "./assets/adaptive-icon.png",
				"backgroundColor": "#FFFFFF"
			}
		},
		"web": {
			"favicon": "./assets/favicon.png"
		}
	}
}
