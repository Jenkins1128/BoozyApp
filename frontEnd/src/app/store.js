import { configureStore } from '@reduxjs/toolkit';
import signUpReducer from '../screens/SignUp/redux/signUpSlice';
import loginRedcuer from '../screens/LogIn/redux/loginSlice';
import homeReducer from '../screens/Home/redux/homeSlice';
import viewRestaurantsReducer from '../screens/Home/redux/viewRestaurantsSlice';
import getDataFromFilterReducer from '../screens/Home/redux/getDataFromFilterSlice';
import restaurantPageReducer from '../screens/RestaurantPage/redux/restaurantPageSlice';
import starRatingReducer from '../screens/RestaurantPage/redux/starRatingSlice';
import menuItemReducer from '../screens/RestaurantPage/redux/menuItemSlice';
import favoriteReducer from '../screens/RestaurantPage/redux/favoriteSlice';
import profileReducer from '../screens/Profile/redux/profileSlice';
import settingsReducer from '../screens/Settings/redux/settingsSlice';

export const store = configureStore({
	reducer: {
		signup: signUpReducer,
		login: loginRedcuer,
		home: homeReducer,
		restaurant: viewRestaurantsReducer,
		dataFromFilter: getDataFromFilterReducer,
		restaurantPage: restaurantPageReducer,
		starRating: starRatingReducer,
		menuItem: menuItemReducer,
		favorite: favoriteReducer,
		profile: profileReducer,
		settings: settingsReducer
	}
});
