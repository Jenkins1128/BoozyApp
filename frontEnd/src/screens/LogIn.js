import React, {Component} from 'react';
import {Animated, StyleSheet, Text, View, TextInput,  TouchableOpacity, ImageBackground, Image } from 'react-native';
//import Todo from './Todo';
// import TodoList from './TodoList';
var isHidden = true;

export default class LogIn extends React.Component{
    
    constructor(props){
        super(props);
        this.state = {
            email: '',
            password: '',
            bounceValue: new Animated.Value(50)
        } 
    }
    

    render(){
        
        return (
            
                <View style={styles.container}>
                      
                        <ImageBackground source={require('./background.jpeg')} style={styles.backgroundImage} >
                        </ImageBackground>
                        <Image source={require('./logo.png')} style={styles.logoImage} >
                        </Image>
                        <Text style={styles.BoozyText}> </Text>
                        <Text style={styles.BoozySlogan}> </Text>
                        <TextInput
                            style={styles.emailInput}
                            onChangeText={(email) => this.setState({email})}
                            value={this.state.email}
                            placeholder='Email'
                            placeholderTextColor='#FFFFFF'
                            underlineColorAndroid='transparent'>
                        </TextInput> 
                        <TextInput
                            style={styles.passwordInput}
                            secureTextEntry={true}
                            onChangeText={(password) => this.setState({password})}
                            value={this.state.password}
                            placeholder='Password'
                            placeholderTextColor='#FFFFFF'
                            underlineColorAndroid='transparent'>
                        </TextInput>     
                        {/* onPress={this.signup.bind(this)} */}
                        <TouchableOpacity onPress={this.signup.bind(this)} style={styles.signUpButton}>
                
                            <Text style={styles.signUpText}>Sign Up</Text>
                        </TouchableOpacity>

                        <TouchableOpacity onPress={this.login.bind(this)} style={styles.loginButton}>
                
                            <Text style={styles.logInText}>Log In</Text>
                        </TouchableOpacity>
                        <Animated.View style={[styles.subView,{transform: [{translateY: this.state.bounceValue}]}]} >
                            <Text style={{fontSize: 20, fontWeight: 'bold', color:"red"}}>Username or password is incorrect.</Text>
                        </Animated.View> 
                    
                </View>
                

        );

    }
    signup(){
        if(!isHidden){
            this.showErrorOverlay();
        }
        this.props.navigation.navigate('SignUp');
    }

   

    login(){
        
        const axios = require('axios').default;
        
        axios.post(`https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/login`, {
                email: this.state.email,
                password: this.state.password
              }, {
                headers: {
                    'Accept': 'application/json, text/plain, */*',
                    'Content-Type': 'application/json'
                }
              }).then((response) => {
                
                //console.log("Logged In!")
                //console.log(response);
                this.goToHome();
              })
              .catch((error) => {
                console.log(error);
                if(isHidden){
                    this.showErrorOverlay();
                }
              });
        
           
              
            //   this.props.navigation.navigate('TodoList', {
            //     dataId: this.state.todoListId,
            //     dataTitle: this.state.todoListTitle,
            //     dataArr: this.state.todoListArray
            //   });
    
    }

    showErrorOverlay(){
        var toValue = 400;
    
        if(isHidden) {
          toValue = -150;
        }
        //This will animate the transalteY of the subview between 0 & 100 depending on its current state
        //100 comes from the style below, which is the height of the subview.
        Animated.spring(
          this.state.bounceValue,
          {
            toValue: toValue,
            velocity: 5,
            tension: 2,
            friction: 8,
          }
        ).start();
    
        isHidden = !isHidden;
    }

    goToHome(){
        if(!isHidden){
            this.showErrorOverlay();
        }
        this.props.navigation.navigate('Home');
    }

}




const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#EB8873'
  },
  header: {
        backgroundColor: '#EB8873',
        justifyContent: 'center',
        alignItems: 'center',
        borderBottomWidth: 10,
        borderBottomColor: '#ddd'
    },
    subView: {
        position: "absolute",
  
        bottom: 0,
        left: 0,
        right: 0,
        backgroundColor: "#FFFFFF",
        height: 30,
        alignItems: 'center',
        
    },
    headerText: {
        backgroundColor: '#fc0',
        width: '100%', // applied to Image
        height: '100%' 
    },
    backgroundImage: {
        flex: 1,
        width: '100%',
        height: '100%',
        justifyContent: "center",
        alignItems: "center",
        opacity: 0.95
    },
    logoImage: {
        position: 'absolute',
        // flex: 1,
        top: 125,
        width: '30%',
        height: '20%',
        justifyContent: "center",
        alignSelf: "center",
        opacity: 0.7
    },
    BoozyText:{
        color: '#FFA500',
        fontSize: 90,
        fontFamily: 'Bradley Hand',
        position: 'absolute',
        top: 125,
        fontWeight: 'bold',
        alignSelf: 'center',
    },
    BoozySlogan:{
        color: '#E91E63',
        fontSize: 24,
        fontFamily: 'Bradley Hand',
        position: 'absolute',
        top: 340,
        fontWeight: 'bold',
        alignSelf: 'center',
    },
    scrollContainer: {
        flex: 1,
        marginBottom: 100
    },
    footer: {
        position: 'absolute',
        bottom: 0,
        left: 0,
        right: 0,
        zIndex: 10
    },
    emailInput: {
        position: 'absolute',
        zIndex: 11,
        top: 350,
        width: 320,
        borderRadius: 30,
        alignSelf: 'center',
        color: '#FFFFFF',
        padding: 30,
        borderWidth: 1,
        borderColor: '#ededed',
    },
    passwordInput: {
        position: 'absolute',
        zIndex: 11,
        top: 440,
        width: 320,
        borderRadius: 30,
        alignSelf: 'center',
        color: '#FFFFFF',
        padding: 30,
        borderWidth: 1,
        borderColor: '#ededed',
    },
    signUpButton: {
        position: 'absolute',
        zIndex: 11,
        top: 540,
        width: 150,
        height: 55,
        borderRadius: 70,
        alignItems: 'center',
        alignSelf: 'center',
        justifyContent: 'center',
        elevation: 8,
        color: '#fff',
        padding: 20,
        backgroundColor: '#FFFFFF',
        
    },
    loginButton: {
        position: 'absolute',
        zIndex: 11,
        top: 610,
        width: 240,
        height: 60,
        borderRadius: 70,
        alignItems: 'center',
        alignSelf: 'center',
        justifyContent: 'center',
        elevation: 8,
        color: '#fff',
        padding: 20,
        backgroundColor: '#FFFFFF',
        
    },
    logInText: {
        color: '#EB8873',
        fontSize: 14,
        fontWeight: 'bold'
    },
    signUpText: {
        color: '#EB8873',
        fontSize: 11,
        fontWeight: 'bold'
    }
});
