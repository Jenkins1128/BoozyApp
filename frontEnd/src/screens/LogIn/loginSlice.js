import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const loginAsync = createAsyncThunk('loginAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.post(data.url, {
			email: data.email,
			password: data.password
		});
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

const loginSlice = createSlice({
	name: 'loginAsync',
	initialState,
	reducers: {
		updateEmail: (state, { payload }) => {
			state.currentState = { ...state.currentState, email: payload.email };
		},
		updatePassword: (state, { payload }) => {
			state.currentState = { ...state.currentState, password: payload.password };
		},
		reset: (state) => {
			state.currentState = {};
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(loginAsync.pending, (state) => {
				state.currentState = { ...state.currentState, requestStatus: 'pending', isLoading: true };
			})
			.addCase(loginAsync.fulfilled, (state) => {
				state.currentState = { ...state.currentState, requestStatus: 'fulfilled', isLoading: false };
			})
			.addCase(loginAsync.rejected, (state) => {
				state.currentState = { ...state.currentState, requestStatus: 'rejected', isLoading: false };
			});
	}
});

export const { updateEmail, updatePassword, reset } = loginSlice.actions;
export const selectLogInState = (state) => state.login.currentState;

export default loginSlice.reducer;
