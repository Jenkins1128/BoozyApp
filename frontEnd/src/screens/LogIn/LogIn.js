import React, { useEffect } from 'react';
import { View, StyleSheet, ImageBackground, TouchableWithoutFeedback, Keyboard, Alert } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { updateEmail, updatePassword, resetStatus, loginAsync, selectLogInState } from './loginSlice';
import background from '../../images/background.jpeg';

import LogInHeader from './LogInHeader';
import LogInInput from './LogInInput';

const LogIn = ({ navigation }) => {
	const state = useSelector(selectLogInState);
	const dispatch = useDispatch();

	useEffect(() => {
		if (!isEmpty(state)) {
			if (state.requestStatus === 'fulfilled') {
				dispatch(resetStatus());
				goToHome();
			} else if (state.requestStatus === 'rejected') {
				dispatch(resetStatus());
				showErrorAlert('Username or password is incorrect.');
			}
		}
	}, [state]);

	const isEmpty = (currentState) => {
		for (const x in currentState) {
			return false;
		}
		return true;
	};

	const showErrorAlert = (errorString) => {
		Alert.alert('Uh oh', errorString, [{ text: 'OK' }]);
	};

	const goToHome = () => {
		navigation.navigate('Home');
	};

	const signup = () => {
		navigation.navigate('SignUp');
	};

	const login = () => {
		if (!isEmpty(state) && state.email.trim().length > 0 && state.password.trim().length > 0) {
			const data = {
				url: 'https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/login',
				email: state.email,
				password: state.password
			};
			dispatch(loginAsync(data));
		} else {
			showErrorAlert('Please fill out all the fields.');
		}
	};

	return (
		<TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
			<View style={styles.container}>
				<ImageBackground source={background} style={styles.backgroundImage}>
					<LogInHeader state={state} />
					<LogInInput state={state} dispatch={dispatch} updateEmail={updateEmail} updatePassword={updatePassword} signup={signup} login={login} />
				</ImageBackground>
			</View>
		</TouchableWithoutFeedback>
	);
};

const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: 'center',
		alignItems: 'center'
	},
	backgroundImage: {
		flex: 1,
		width: '100%',
		height: '100%',
		opacity: 0.95
	}
});

export default LogIn;
