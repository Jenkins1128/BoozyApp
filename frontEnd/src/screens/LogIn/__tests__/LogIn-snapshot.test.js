import React from 'react';
import { Provider } from 'react-redux';
import { store } from '../../../app/store';

import renderer from 'react-test-renderer';
import LogIn from '../LogIn';

test('LogIn renders correctly', () => {
	const tree = renderer
		.create(
			<Provider store={store}>
				<LogIn />
			</Provider>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
