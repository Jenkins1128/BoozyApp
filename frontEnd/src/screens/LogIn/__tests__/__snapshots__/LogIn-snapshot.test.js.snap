// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 1,
          "height": "100%",
          "opacity": 0.95,
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/background.jpeg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 2,
            "marginTop": 100,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/logo.png",
            }
          }
          style={
            Object {
              "height": "40%",
              "opacity": 0.7,
              "width": "30%",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 4,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Email"
          placeholderTextColor="#FFFFFF"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "borderColor": "#ededed",
              "borderRadius": 30,
              "borderWidth": 1,
              "color": "#FFFFFF",
              "marginBottom": 10,
              "padding": 30,
              "width": 320,
            }
          }
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="#FFFFFF"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "alignSelf": "center",
              "borderColor": "#ededed",
              "borderRadius": 30,
              "borderWidth": 1,
              "color": "#FFFFFF",
              "marginBottom": 10,
              "padding": 30,
              "width": 320,
            }
          }
          underlineColorAndroid="transparent"
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 70,
              "color": "#fff",
              "elevation": 8,
              "height": 55,
              "justifyContent": "center",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 20,
              "width": 150,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#EB8873",
                "fontSize": 11,
                "fontWeight": "bold",
              }
            }
          >
            Sign Up
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 70,
              "color": "#fff",
              "elevation": 8,
              "height": 60,
              "justifyContent": "center",
              "opacity": 1,
              "padding": 20,
              "width": 240,
            }
          }
          testID="loginButton"
        >
          <Text
            style={
              Object {
                "color": "#EB8873",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            Log In
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
